app/controllers/concerns/course/forum/topic_controller_subscription_concern.rb

```diff
@@ -0,0 +1,37 @@
+module Course::Forum::TopicControllerSubscriptionConcern
+  extend ActiveSupport::Concern
+
+  def subscribe
+    if set_subscription_state
```

Comment:
Okay, but I don't understand why it's safer ? Subscribe is at the show page, think if user can see the topic then there is not need to especially check for subscribe ?